components/odh-notebook-controller/controllers/notebook_dspa_secret.go

```diff
@@ -0,0 +1,347 @@
+	}
+
+	// Extract API Endpoint from DSPA status
+	apiEndpoint := dspaInstance.Status.Components.APIServer.ExternalUrl
+
+	// Extract info from DSPA spec
+	spec := dspaInstance.Spec
+	objectStorage := spec.ObjectStorage
+	externalStorage := objectStorage.ExternalStorage
+
+	// Validate required fields
+	host := externalStorage.Host
+	if host == "" {
+		return nil, fmt.Errorf("invalid DSPA CR: missing or invalid 'host'")
+	}
+	cosEndpoint := fmt.Sprintf("https://%s", host)
+
+	cosBucket := externalStorage.Bucket
+	if cosBucket == "" {
+		return nil, fmt.Errorf("invalid DSPA CR: missing or invalid 'bucket'")
+	}
+
+	s3CredentialsSecret := externalStorage.S3CredentialSecret
+	cosSecret := s3CredentialsSecret.SecretName
+	usernameKey := s3CredentialsSecret.AccessKey
+	passwordKey := s3CredentialsSecret.SecretKey
+
+	// Fetch secret containing credentials
+	dspaCOSSecret := &corev1.Secret{}
+	err := client.Get(ctx, types.NamespacedName{Name: cosSecret, Namespace: notebook.Namespace}, dspaCOSSecret)
+	if err != nil {
+		log.Error(err, "Failed to get secret", "secretName", cosSecret)
+		return nil, fmt.Errorf("failed to get secret '%s': %w", cosSecret, err)
+	}
+
+	usernameVal, ok := dspaCOSSecret.Data[usernameKey]
+	if !ok {
+		return nil, fmt.Errorf("missing key '%s' in secret '%s'", usernameKey, cosSecret)
+	}
+	passwordVal, ok := dspaCOSSecret.Data[passwordKey]
+	if !ok {
+		return nil, fmt.Errorf("missing key '%s' in secret '%s'", passwordKey, cosSecret)
+	}
+
+	// Construct Elyra-compatible config
+	return map[string]interface{}{
+		"display_name": "Data Science Pipeline",
+		"schema_name":  "kfp",
+		"metadata": map[string]interface{}{
+			"tags":                []string{},
+			"display_name":        "Data Science Pipeline",
+			"engine":              "Argo",
+			"runtime_type":        "KUBEFLOW_PIPELINES",
+			"auth_type":           "KUBERNETES_SERVICE_ACCOUNT_TOKEN",
+			"cos_auth_type":       "KUBERNETES_SECRET",
+			"public_api_endpoint": publicAPIEndpoint,
+			"api_endpoint":        apiEndpoint,
+			"cos_endpoint":        cosEndpoint,
+			"cos_bucket":          cosBucket,
+			"cos_username":        string(usernameVal),
+			"cos_password":        string(passwordVal),
+			"cos_secret":          cosSecret,
+		},
+	}, nil
+}
+
+// NewElyraRuntimeConfigSecret defines and handles the creation, watch and update to the desired ElyraRuntimeConfig secret object
+func (r *OpenshiftNotebookReconciler) NewElyraRuntimeConfigSecret(ctx context.Context, dynamicConfig *rest.Config, c client.Client, notebook *nbv1.Notebook, controllerNamespace string, log logr.Logger) error {
+	dynamicClient, err := dynamic.NewForConfig(dynamicConfig)
+	if err != nil {
+		log.Error(err, "Failed to create dynamic client")
+		return err
+	}
+	dashboardInstance, err := getDashboardInstance(ctx, dynamicClient, log)
+	if err != nil {
+		return err
+	}
+	// Skipping Elyra secret creation; Dashboard CR not found (optional cr)
+	if dashboardInstance == nil {
+		return nil
+	}
+
```

Comment:
confused on the return statement, 
as dashboard URL is optional for  elyra runtime config setup, 
irrespective of err or empty instance, lets pass through without breaking function with return
as we want to continue creating the secret, even if dashboard URL is not present. 

This can happen, if the Dashboard is not selected in DSC